ext/ManoptJuMPExt.jl

```diff
@@ -0,0 +1,408 @@
+`VectorizedManifold` `set`.
+"""
+function MOI.add_constrained_variables(model::Optimizer, set::VectorizedManifold)
+    F = MOI.VectorOfVariables
+    if !isnothing(model.manifold)
+        throw(
+            AddConstraintNotAllowed{F,typeof(set)}(
+                "Only one manifold allowed, variables in `$(model.manifold)` have already been added.",
+            ),
+        )
+    end
+    model.manifold = set.manifold
+    model.problem = nothing
+    model.state = nothing
+    n = MOI.dimension(set)
+    v = MOI.VariableIndex.(1:n)
+    for _ in 1:n
+        push!(model.variable_primal_start, nothing)
+    end
+    return v, MOI.ConstraintIndex{F,typeof(set)}(1)
+end
+
+"""
+    MOI.is_valid(model::Optimizer, vi::MOI.VariableIndex)
+
+Return whether `vi` is a valid variable index.
+"""
+function MOI.is_valid(model::Optimizer, vi::MOI.VariableIndex)
+    return !isnothing(model.manifold) &&
+           1 <= vi.value <= MOI.dimension(VectorizedManifold(model.manifold))
+end
+
+"""
+    MOI.get(model::Optimizer, ::MOI.NumberOfVariables)
+
+Return the number of variables added in the model, this corresponds
+to the [`MOI.dimension`](@ref) of the `VectorizedManifold`.
+"""
+function MOI.get(model::Optimizer, ::MOI.NumberOfVariables)
+    if isnothing(model.manifold)
+        return 0
+    else
+        return MOI.dimension(VectorizedManifold(model.manifold))
+    end
+end
+
+"""
+    MOI.supports(::Manopt.JuMP_Optimizer, attr::MOI.RawOptimizerAttribute)
+
+Return `true` indicating that `Manopt.JuMP_Optimizer` supports starting values
+for the variables.
+"""
+function MOI.supports(::Optimizer, ::MOI.VariablePrimalStart, ::Type{MOI.VariableIndex})
+    return true
+end
+
+"""
+    function MOI.set(
+        model::Optimizer,
+        ::MOI.VariablePrimalStart,
+        vi::MOI.VariableIndex,
+        value::Union{Real,Nothing},
+    )
+
+Set the starting value of the variable of index `vi` to `value`. Note that if
+`value` is `nothing` then it essentially unset any previous starting values set
+and hence `MOI.optimize!` unless another starting value is set.
+"""
+function MOI.set(
+    model::Optimizer,
+    ::MOI.VariablePrimalStart,
+    vi::MOI.VariableIndex,
+    value::Union{Real,Nothing},
+)
+    MOI.throw_if_not_valid(model, vi)
+    model.variable_primal_start[vi.value] = value
+    model.state = nothing
+    return nothing
+end
+
+function MOI.supports(::Optimizer, ::Union{MOI.ObjectiveSense,MOI.ObjectiveFunction})
+    return true
+end
+
+function MOI.set(model::Optimizer, ::MOI.ObjectiveSense, sense::MOI.OptimizationSense)
+    model.sense = sense
+    return nothing
+end
+
+MOI.get(model::Optimizer, ::MOI.ObjectiveSense) = model.sense
+
+function MOI.get(
+    model::Optimizer, attr::Union{MOI.ObjectiveFunctionType,MOI.ObjectiveFunction}
+)
+    return MOI.get(model.nlp_model, attr)
+end
+
+function MOI.set(model::Optimizer, ::MOI.ObjectiveFunction{F}, func::F) where {F}
```

Comment:
same as before, a small doc strip would be nice, and again the name seems missing the `!`.